### DISCUSSION TITLE
package name customization: exploring the possibilities

### DISCUSSION THREAD

**Discussion by alexanderankin** [casid/jte#301] [Ideas]
starting this as a discussion to highlight the hypothetical nature/intent:

how does this project consider package structuring? I caught a little bit where about interoperability in #278, but my overall concern comes from the fact that when i set up my local gradle environment, i had to prefix "jte/" to the template name. So from my point of view, the lack of customizability of the package name is a bug.

But i want to explore and see what the general feeling about customizing package names is. For example, in the spring boot starter, the package is hard-coded. When i set a custom one in the gradle plugin, i was real confused:

```groovy
def projectDir = layout.projectDirectory.asFile.path // aka __dirname

jte.precompile()
jte.sourceDirectory.set java.nio.file.Path.of(projectDir, 'src', 'main', 'resources')
jte.targetDirectory.set java.nio.file.Path.of(projectDir, 'build', 'generated', 'sources', 'jte', 'java', 'main')
// jte.packageName.set 'proto3.jte'  // not working because starter hardcodes:
// gg.jte.runtime.Constants#PACKAGE_NAME_PRECOMPILED

sourceSets.main.java.srcDir jte.targetDirectory.get()
```

if the package name can be configured in the gradle plugin, then it seems that the same setting should be configurable at runtime in `gg.jte.springframework.boot.autoconfigure.JteProperties` - but i wanted to get a sense of the bigger picture before i just submit a pr if it doesnt align with the needs of the project.

thanks for reading.

----

edit: actually, realized why - i forgot to source them from `resources/jte`, intead of just `resources`. so maybe this is not so bad, but maybe package customization is still welcome as topic? edit again: nope, still no dice - needs `jte/` prefix in controller. edit 3: gradle task is not being ran?? edit - for this approach to work use generate, and make compilation dependent on generation - massive footgun here.

**Comment by edward3h** [comment]
Yes, you can set the package name when using the Gradle plugin. Something like:

```
jte {
    generate()
    packageName = 'com.example.jte'
}
```
I don't use Spring so I can't comment on that part.

**Reply by alexanderankin** [reply]
coming back to it a day later and this does indeed work, i guess i was more confused by the generate vs precompile than i thought
